packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol

```diff
@@ -174,31 +179,47 @@ contract TokenFacet is BaseConnextFacet {
     address _stableSwapPool,
     uint256 _cap
   ) external onlyOwnerOrAdmin returns (address _local) {
+    // Calculate the canonical key.
+    bytes32 key = AssetLogic.calculateCanonicalHash(_canonical.id, _canonical.domain);
+
     bool onCanonical = _canonical.domain == s.domain;
-    if (!onCanonical) {
-      // On remote, deploy a local representation
-      _local = _deployRepresentation(
-        _canonical.id,
-        _canonical.domain,
-        _canonicalDecimals,
-        _representationName,
-        _representationSymbol
-      );
-    } else {
+    if (onCanonical) {
+      if (_cap > 0) {
+        // Set caps only on canonical domain
+        _setLiquidityCap(_canonical, _cap, key);
+      }
+
       // Get the local address
       _local = TypeCasts.bytes32ToAddress(_canonical.id);
 
       // You are on the canonical domain, ensure the adopted asset is empty
       if ((_adoptedAssetId != _local && _adoptedAssetId != address(0)) || _stableSwapPool != address(0)) {
         revert TokenFacet__setupAsset_invalidCanonicalConfiguration();
       }
+    } else {
+      // Cannot already have an assigned representation.
+      // NOTE: *If* it does, it can still be replaced with `setupAssetWithDeployedRepresentation`
+      if (s.canonicalToRepresentation[key] != address(0)) {
+        revert TokenFacet__setupAsset_representationListed();
+      }
+      // On remote, deploy a local representation.
+      _local = _deployRepresentation(
+        _canonical.id,
+        _canonical.domain,
+        _canonicalDecimals,
+        _representationName,
+        _representationSymbol
+      );
     }
 
     // Enroll the asset
     bytes32 key = _enrollAdoptedAndLocalAssets(_adoptedAssetId, _local, _stableSwapPool, _canonical);
```

Comment:
`_enrollAdoptedAndLocalAssets` has already been called below and does not return any values. This line can be removed.